File: front/src/app/models/curso.ts
export class Curso {
	public id_curso: number;

	public nombre_curso: string;

	public profesores_curso: number[];

	public fecha_publicacion_curso: Date;

	public clases_curso: number[];

	public planes_curso: number[];

	public precio_curso: number;

	constructor(_id_curso: number, _nombre_curso: string, _profesores_curso: number[], _fecha_publicacion_curso: Date, _clases_curso: number[], _planes_curso: number[], _precio_curso: number) {
		this.id_curso = _id_curso;
		this.nombre_curso = _nombre_curso;
		this.profesores_curso = _profesores_curso;
		this.fecha_publicacion_curso = _fecha_publicacion_curso;
		this.clases_curso = _clases_curso;
		this.planes_curso = _planes_curso;
		this.precio_curso = _precio_curso;
	}
}

File: back_streaming/src/main/java/com/sovereingschool/back_streaming/Models/Curso.java
package com.sovereingschool.back_streaming.Models;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Entity
@Table(name = "curso")
public class Curso implements Serializable {
    @Id
    private Long id_curso;

    private String nombre_curso;

    private List<Usuario> profesores_curso;

    private Date fecha_publicacion_curso;

    private List<Clase> clases_curso;

    private List<Plan> planes_curso;

    private BigDecimal precio_curso;

}